js/src/components/tree-select-control/index.js

```diff
@@ -296,8 +315,13 @@ const TreeSelectControl = ( {
 	};
 
 	return (
+		// eslint-disable-next-line jsx-a11y/no-static-element-interactions
 		<div
 			{ ...focusOutside }
+			onKeyDown={ onKeyDown }
+			onClick={ () => {
+				setTreeVisible( true );
+			} }
```

Comment:
Why did we add the onClick? 

This is because of the Escape feature. When you click escape, the Tree collapses but the input still has the focus. Hence, if you click on it, no onFocus event is triggered anymore and the tree doesn't expand.

Why we didn't remove the onFocus? 

On the other side, if we remove the onFocus, when the user navigates over the elements with Tab and the Tree Select gets the focus, it will not open since onFocus is not there anymore. 

Having both is the only way I found so far, open to fresh ideas tho...

Reply:
Thanks for the clarification! I thought about it, probably both event handlers are necessary and cannot omit either one.